File: backend/app/schemas/food_log.py
from datetime import datetime
from typing import Literal
from pydantic import BaseModel

MealType = Literal["breakfast", "lunch", "dinner", "snack"]


class FoodLogCreate(BaseModel):
    name: str
    calories: int
    protein_g: float
    carbs_g: float
    fat_g: float
    meal_type: MealType


class FoodLogUpdate(BaseModel):
    name: str | None = None
    calories: int | None = None
    protein_g: float | None = None
    carbs_g: float | None = None
    fat_g: float | None = None
    meal_type: MealType | None = None


class FoodLogParseRequest(BaseModel):
    text: str


class FoodLogParseResponse(BaseModel):
    name: str | None = None
    calories: int | None = None
    protein_g: float | None = None
    carbs_g: float | None = None
    fat_g: float | None = None
    meal_type: MealType | None = None


class FoodLogOut(BaseModel):
    id: int
    name: str
    calories: int
    protein_g: float
    carbs_g: float
    fat_g: float
    meal_type: MealType
    logged_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/foodLog.ts
export type MealType = "breakfast" | "lunch" | "dinner" | "snack";

export interface FoodLogCreate {
  name: string;
  calories: number;
  protein_g: number;
  carbs_g: number;
  fat_g: number;
  meal_type: MealType;
}

export interface FoodLogParseResponse {
  name: string | null;
  calories: number | null;
  protein_g: number | null;
  carbs_g: number | null;
  fat_g: number | null;
  meal_type: MealType | null;
}

export interface FoodLogOut {
  id: number;
  name: string;
  calories: number;
  protein_g: number;
  carbs_g: number;
  fat_g: number;
  meal_type: MealType;
  logged_at: string;
}
